### PR TITLE
进度检查 - 正确计算同网段内对等体的总上传量

### PR DESCRIPTION
**Pull request by paulzzh**
替代 #319 
progressRecorder仍以网段为key，但包含同网段所有的ClientTask
db内持久化储存ip，不保存网段
需要近一步测试

## Summary by CodeRabbit

- **New Features**
	- Introduced a method for checking and updating IPv6 prefix length configuration.
	- Enhanced handling of IP addresses for client task management.

- **Bug Fixes**
	- Improved logic for managing maximum weight on specific operations related to IP addresses.

- **Documentation**
	- Updated configuration parameters affecting network behavior and application versioning.

### CODE REVIEW

**Review by Ghost-chu — CHANGES_REQUESTED** [review]

File: src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/ProgressCheatBlocker.java [hunk 1]
```diff
@@ -194,19 +194,22 @@ private void reloadConfig() {
...
+        // 命中缓存但缓存中无当前种子 需要从数据库导入
+        if (lastRecordedProgress.stream().noneMatch(task -> task.getTorrentId().equals(torrent.getId()))){
+            lastRecordedProgress.addAll(0,loadClientTasks(peerPrefix, torrent.getId()));
         }
-        if (clientTask == null) {
+        ClientTask clientTask;
+        // 需要满足同个ip和种子
+        Optional<ClientTask> selected = lastRecordedProgress.stream().filter(task -> task.getPeerIp().equals(peerIpString) && task.getTorrentId().equals(torrent.getId())).findFirst();
+        if (selected.isPresent()) {
+            clientTask = selected.get();
```
[the remark this answers is not in the document]
Comment (paulzzh) [reply src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/ProgressCheatBlocker.java]: 改为默认参数null，后续判断，无法将new ClientTask()作为默认参数

File: src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/ProgressCheatBlocker.java [hunk 2]
```diff
@@ -333,6 +329,13 @@ public static class ClientTask {
         private long firstTimeSeen;
         private long lastTimeSeen;
     }
+
+    @AllArgsConstructor
+    @Data
+    public static class Client {
+        private String peerPrefix;
+        private String torrentId;
+    }
```
Comment (paulzzh) [inline src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/ProgressCheatBlocker.java:338]: @coderabbitai Lombok's `@Data` has already override `equals` and `hashCode`.

File: src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/ProgressCheatBlocker.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/ProgressCheatBlocker.java]: 有额外操作可以用 orElseGet，在 Supplier 里面执行额外操作。

[the remark this answers is not in the document]
Comment (paulzzh) [reply src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/ProgressCheatBlocker.java]: Supplier里面仍然lambda需要将new ClientTask()添加进lastRecordedProgress，要求lastRecordedProgress为final，上面有进行重新赋值，不太方便
```
        ).findFirst().orElseGet(
                () -> {
                    ClientTask task = new ClientTask(peerIpString, torrent.getId(), 0d, 0L, 0L, 0, 0, System.currentTimeMillis(), System.currentTimeMillis());
                    lastRecordedProgress.add(task);
                }
        );
```

**Pull request merged**